src/pu-log.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ */
+
+#include <glib.h>
+#include "pu-log.h"
+
+#define PU_LOG_DOMAINS "partup partup-config partup-emmc partup-mount partup-utils"
+
+GLogLevelFlags log_output_level;
+
+static GLogWriterOutput
+pu_log_writer(GLogLevelFlags log_level,
+              const GLogField *fields,
+              gsize n_fields,
+              gpointer user_data)
+{
+    const gchar *log_domain = NULL;
+
+    for (gsize i = 0; i < n_fields; i++) {
+        if (g_strcmp0(fields[i].key, "GLIB_DOMAIN") == 0) {
+            log_domain = fields[i].value;
+            break;
+        }
+    }
+
+    if (log_level > log_output_level ||
+        g_log_writer_default_would_drop(log_level, log_domain))
```

Comment:
That looks confusing in conjunction with the oneliner below. Please merge these two lines. It's still under 100 characters wide, so should be fine.